tasks/verify-protocol.ts

```diff
@@ -0,0 +1,209 @@
+import {task} from "hardhat/config"
+import {HardhatRuntimeEnvironment} from "hardhat/types"
+import {ethers} from "ethers"
+import getNetworkConfig from "../deploy/migrations.config"
+
+import {
+    BondingManager,
+    Controller as ControllerContract,
+    Governor,
+    Minter,
+    RoundsManager,
+    TicketBroker
+} from "../typechain"
+
+const BridgeContracts = {
+    L2Migrator: "0x148D5b6B4df9530c7C76A810bd1Cdf69EC4c2085",
+    L2MigratorTarget: "0x4F59b39e2ea628fe8371BDfd51B063319339c7EE"
+}
+
+class Controller {
+    public controller: ControllerContract
+    public deployments: any
+
+    constructor(controller: ControllerContract, deployments: any) {
+        this.controller = controller
+        this.deployments = deployments
+    }
+
+    get address() {
+        return this.controller.address
+    }
+
+    static async new(hre: HardhatRuntimeEnvironment) {
+        // @ts-ignore
+        const {deployments, ethers} = hre
+        const controllerDeployment = await deployments.get("Controller")
+        const controller: ControllerContract = await ethers.getContractAt(
+            "Controller",
+            controllerDeployment.address
+        )
+
+        return new Controller(controller, deployments)
+    }
+
+    static getContractId(name: string) {
+        return ethers.utils.solidityKeccak256(["string"], [name])
+    }
+
+    async getContractAddr(name: string) {
+        return this.controller.getContract(Controller.getContractId(name))
+    }
+
+    async verifyRegisteredContract(name: string) {
+        const registeredAddress = await this.getContractAddr(name)
+
+        let deployedAddress: string
+        if (name === "L2Migrator") {
+            deployedAddress = BridgeContracts.L2Migrator
+        } else if (name === "L2MigratorTarget") {
+            deployedAddress = BridgeContracts.L2MigratorTarget
```

Comment:
Can be improved, this was the simplest implementation for now. Same for creating `BridgeContracts` object.